include/Basis/CartesianExponents.hpp

```diff
@@ -0,0 +1,120 @@
+// This file is part of GQCG-gqcp.
+// 
+// Copyright (C) 2017-2019  the GQCG developers
+// 
+// GQCG-gqcp is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// GQCG-gqcp is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-gqcp.  If not, see <http://www.gnu.org/licenses/>.
+// 
+#ifndef CartesianExponents_hpp
+#define CartesianExponents_hpp
+
+
+#include "CartesianDirection.hpp"
+
+#include <array>
+#include <vector>
+
+
+namespace GQCP {
+
+
+/**
+ *  A class that represents exponents of the Cartesian functions x, y and z
+ */
+class CartesianExponents {
+private:
+    size_t x_exponent;
+    size_t y_exponent;
+    size_t z_exponent;
+
+
+public:
+    // CONSTRUCTORS
+    /**
+     *  @param x        the exponent in x
+     *  @param y        the exponent in y
+     *  @param z        the exponent in z
+     */
+    CartesianExponents(size_t x, size_t y, size_t z);
+
+    /**
+     *  @param arr      the array containing the x-, y- and z-exponent in that order
+     */
+    CartesianExponents(const std::array<size_t, 3>& arr);
+
+
+    // OPERATORS
+    /**
+     *  @param rhs      the right-hand side of the operator <
+     *
+     *  @return if these Cartesian exponents are 'smaller' than the ones on the right-hand side. The following logic is used: lhs < rhs
+     *      - if lhs's angular momentum is smaller
+     *      - if both angular momenta are equal, x takes precedence over y, over z
+     *
+     *  This means that {1, 0, 0}(=x) < {2, 0, 0}(=x^2), and {2, 0, 0}(=x^2) < {1, 1, 0}(=xy)
+     */
+    bool operator<(const CartesianExponents& rhs) const;
+
+    /**
+     *  @param rhs      the right-hand side of the operator ==
+     *
+     *  @return if the Cartesian exponents are considered equal
+     */
+    bool operator==(const CartesianExponents& rhs) const;
+
+    /**
+     *  @param rhs      the right-hand side of the operator ==
+     *
+     *  @return if the Cartesian exponents are considered different
+     */
+    bool operator!=(const CartesianExponents& rhs) const;
+
+
+    // GETTERS
+    size_t x() const { return this->x_exponent; }
+    size_t y() const { return this->y_exponent; }
+    size_t z() const { return this->z_exponent; }
+    size_t& x() { return this->x_exponent; }
```

Comment:
The `size& ...()`-methods aren't really getters, I guess these are more like setters. 

I've made an update to `CartesianExponents`, so that it is now a struct instead of a class, so that these methods can be removed.